core/chains/evm/client/node.go

```diff
@@ -288,10 +288,11 @@ func (n *node) verify(callerCtx context.Context) (err error) {
 		promEVMPoolRPCNodeVerifiesFailed.WithLabelValues(n.chainID.String(), n.name).Inc()
 	}
 
-	switch n.state {
+	st := n.State()
```

Comment:
wait, this should always be done inside a mutex. Is there a case where it is not? See comment on line 280

Reply:
None of the callers are locking the mutex. In the one case of `start` that was fine, but he others were racey and one was flagged by the detector in a local run which led me here.
Most of the callers make other calls to methods which lock themseles, including the corresponding `State()` method which does it's own locking, so this seemed to fit local style but I overlooked the comment. Feel free to fix either way, but FWIW locking as tightly as possible is usually favorable.

Reply:
Yup, looks reasonable